utils/networking/src/resolver.rs

```diff
@@ -71,12 +71,10 @@ pub async fn try_resolve_dns_record(request_url_or_host: &str) -> String {
     lazy_static! {
         static ref SCHEME_RE: Regex = Regex::new("(?i)^[a-z0-9\\-\\.]+?:").unwrap();
     }
-    match {
-        if SCHEME_RE.is_match(request_url_or_host) {
-            resolve_dns_record(request_url_or_host).await
-        } else {
-            resolve_dns_record_host(request_url_or_host).await
-        }
+    match if SCHEME_RE.is_match(request_url_or_host) {
+        resolve_dns_record(request_url_or_host).await
+    } else {
+        resolve_dns_record_host(request_url_or_host).await
```

Comment:
can you extract the `if { .. } else { .. }` into a variable? It's not reallly readable without the previously present brackets